### PR TITLE
Feat/DEVSU-2301 My Reports View

### PR DESCRIPTION
**Pull request by bnguyen-bcgsc**
- Create new view for My Reports
- Add new route to Main View
- Add new access point to side bar with reworked Reports section to drop down menu
- Update sidebar unit testing to account for new My Reports view
- Rewrite Reports Table as a separate component for future reusing (when creating new filtered reports views)
- Remove PatientsView component and all its subsequent routing
- Code linting

### CODE REVIEW

File: app/views/ReportsView/index.tsx [hunk 1]
```diff
@@ -80,43 +67,9 @@ const ReportsTableComponent = (): JSX.Element => {
...
-        }}
-        onGridReady={onGridReady}
-        onGridSizeChanged={onGridSizeChanged}
-        pagination
-        paginationAutoPageSize
-        rowData={rowData}
-        rowSelection="single"
-      />
-    </div>
+    <>{ReportsTableComponent(rowData)}</>
```
Comment (kttkjl) [inline app/views/ReportsView/index.tsx:71]: `<ReportsTableComponent props=(rowData)}>` ?

though, I'd do `<ReportsTableComponent rowData=(rowData)}>`

File: app/components/ReportsTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import LaunchCell from '@/components/LaunchCell';
+import NoRowsOverlay from '@/components/DataTable/components/NoRowsOverlay';
+import columnDefs from './columnDefs';
+
+import './index.scss';
+
+/**
+ * Report table containing all reports
+ */
+const ReportsTableComponent = (reportsData: ReportType[]): JSX.Element => {
```
Comment (kttkjl) [inline app/components/ReportsTable/index.tsx:17]: more of a chore, but shouldn't we destructure props into named ones?

**Review by kttkjl — COMMENTED** [review]
Looks good, but just some React conventions should be followed

File: app/components/ReportsTable/index.tsx (hunk 2 above)
Comment (bnguyen-bcgsc) [reply app/components/ReportsTable/index.tsx:17]: Hmm when I do that I got this error in React: Cannot destructure property 'rowData' of '_ref' as it is undefined

**Review by kttkjl — APPROVED** [review]

**Pull request merged**